boa/src/syntax/parser/mod.rs

```diff
@@ -876,41 +1015,1330 @@ impl Parser {
+            )),
+            _ => {
+                self.step_back();
+                self.read_update_expression()
+            }
+        }
+    }
+
+    /// <https://tc39.es/ecma262/#prod-UpdateExpression>
+    fn read_update_expression(&mut self) -> Result<Node, ParseError> {
+        let tok = self.peek_skip_lineterminator()?;
+        match tok.kind {
+            TokenKind::Punctuator(Punctuator::Inc) => {
+                self.next_skip_lineterminator().unwrap();
+                return Ok(Node::UnaryOp(
+                    UnaryOp::IncrementPre,
+                    Box::new(self.read_left_hand_side_expression()?),
+                ));
+            }
+            TokenKind::Punctuator(Punctuator::Dec) => {
+                self.next_skip_lineterminator().unwrap();
+                return Ok(Node::UnaryOp(
+                    UnaryOp::DecrementPre,
+                    Box::new(self.read_left_hand_side_expression()?),
+                ));
+            }
+            _ => {}
+        }
+
+        let lhs = self.read_left_hand_side_expression()?;
+        if let Ok(tok) = self.peek(0) {
+            match tok.kind {
+                TokenKind::Punctuator(Punctuator::Inc) => {
+                    self.get_next_token().unwrap();
+                    return Ok(Node::UnaryOp(UnaryOp::IncrementPost, Box::new(lhs)));
+                }
+                TokenKind::Punctuator(Punctuator::Dec) => {
+                    self.get_next_token().unwrap();
+                    return Ok(Node::UnaryOp(UnaryOp::DecrementPost, Box::new(lhs)));
+                }
+                _ => {}
+            }
+        }
+
+        Ok(lhs)
+    }
+
+    /// https://tc39.github.io/ecma262/#prod-LeftHandSideExpression
+    /// TODO: Implement NewExpression: new MemberExpression
+    fn read_left_hand_side_expression(&mut self) -> Result<Node, ParseError> {
+        let lhs = self.read_member_expression()?;
+        match self.peek_skip_lineterminator() {
+            Ok(ref tok) if tok.kind == TokenKind::Punctuator(Punctuator::OpenParen) => {
+                self.read_call_expression(lhs)
+            }
+            _ => self.read_new_expression(lhs),
+        }
+    }
+
+    /// https://tc39.github.io/ecma262/#prod-NewExpression
+    fn read_new_expression(&mut self, first_member_expr: Node) -> Result<Node, ParseError> {
+        Ok(first_member_expr)
+    }
+
+    /// <https://tc39.es/ecma262/#prod-MemberExpression>
+    fn read_member_expression(&mut self) -> Result<Node, ParseError> {
+        let mut lhs = if self.peek_skip_lineterminator()?.kind == TokenKind::Keyword(Keyword::New) {
+            self.next_skip_lineterminator()?;
+            let lhs = self.read_member_expression()?;
+            self.expect_punc(Punctuator::OpenParen, "expect '('.");
+            let args = self.read_arguments()?;
+            let call_node = Node::Call(Box::new(lhs), args);
+            let new_node = Node::New(Box::new(call_node));
+            new_node
```

Comment:
This should fix the clippy error:
```suggestion
            Node::New(Box::new(call_node))
```